final-project/application.py

```diff
@@ -498,23 +498,22 @@ def product():
     user_id = session.get("user_id")
 
     # Request a product from the user.
-    product_request = request.args.get("product")
+    id_products_request = request.args.get("id")
```

Comment:
Rather than querying for id, it should be routed to the id directly. People cannot query for specific ids b/c that doesn't make sense from a user perspective.